pkg/app/piped/eventwatcher/eventwatcher.go

```diff
@@ -215,20 +223,37 @@ func (w *watcher) updateValues(ctx context.Context, repo git.Repo, events []conf
 	}
 	defer tmpRepo.Clean()
 
+	handledEventIDs := make([]string, 0, len(events))
 	for _, e := range events {
-		if err := w.commitFiles(ctx, e, tmpRepo, commitMsg); err != nil {
+		id, err := w.commitFiles(ctx, e, tmpRepo, commitMsg)
+		if err != nil {
 			w.logger.Error("failed to commit outdated files", zap.Error(err))
 			continue
 		}
+		if id != "" {
+			handledEventIDs = append(handledEventIDs, id)
+		}
+	}
+	if err := tmpRepo.Push(ctx, tmpRepo.GetClonedBranch()); err != nil {
+		return fmt.Errorf("failed to push commits: %w", err)
 	}
-	return tmpRepo.Push(ctx, tmpRepo.GetClonedBranch())
+	for _, id := range handledEventIDs {
+		if _, err := w.apiClient.ReportEventHandled(ctx, &pipedservice.ReportEventHandledRequest{EventId: id}); err != nil {
+			return fmt.Errorf("failed to report that event %q is handled: %w", id, err)
+		}
+	}
```

Comment:
Or we can update the RPC to mark a list of events instead of only one.

Reply:
Sounds better. It needs only event-id, so will not get complicated that much.